internal/app/fipcontroller/kubernetes.go

```diff
@@ -31,32 +31,79 @@ func newKubernetesClient() (*kubernetes.Clientset, error) {
  * Search and return the IP address of a given kubernetes node name.
  *  Will return first found internal or external IP depending on nodeAddressType parameter
  */
-func (controller *Controller) nodeAddress(ctx context.Context, nodeName string, nodeAddressType configuration.NodeAddressType) (address net.IP, err error) {
-	nodes, err := controller.KubernetesClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+func (controller *Controller) nodeAddressList(ctx context.Context, nodeAddressType configuration.NodeAddressType) (addressList []net.IP, err error) {
+	// Try to get deployment pods if certain label is specified
+	listOptions := metav1.ListOptions{}
+	listOptions.LabelSelector = controller.Configuration.PodLabelSelector
+
+	pods, err := controller.KubernetesClient.CoreV1().Pods(controller.Configuration.Namespace).List(ctx, listOptions)
```

Comment:
* What does `Pods().List()` return with an empty (default?) labelSelector?
* Limiting the List to the controller namespace might make using a user-defined PodLabelSelector behave unexpectedly.

Reply:
Pods().List() will return all pods in the namespace when no labelSelector is specified.

Not limiting the controller to it's own namespace would require it to have a ClusterRoleBinding.
I can add an option to not limit the namespace but it definitely shouldn't be the default.
And I don't really see, how a user-defined PodLabelSelector could behave unexpectedly here, as it should only be used to search for it's own pods.

Reply:
Pods().List() will return all pods from the fip-controller deployment with the default labelSelector, if the label in the default deployment was not changed.
Otherwise it will return no pods at all.